modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/IdentityProviderMgtServiceTestCase.java

```diff
@@ -360,7 +365,202 @@ public void testGetAllIdPs() throws Exception {
         }
     }
 
-    @Test(priority = 4, groups = "wso2.is", description = "test getEnabledAllIdPs operation")
+    @Test(priority = 4, groups = "wso2.is", description = "test getAllPaginatedIdPsInfo operation")
+    public void testGetPaginatedIdPs() throws Exception {
+
+        addIdpForPagination();
+        int pageNumber = 1;
+        List<IdentityProvider> idpList = idpMgtServiceClient.getAllPaginatedIdPsInfo(pageNumber);
+        Assert.assertNotNull(idpList);
+        log.info("All idp paginated list size : " + idpList.size());
+        if (idpList.size() > 0) {
+            IdentityProvider filteredIdp = null;
+            for (IdentityProvider idp : idpList) {
+                if (testIdpName.equals(idp.getIdentityProviderName()) || testIdpNameSearch
+                        .equals(idp.getIdentityProviderName())) {
+                    filteredIdp = idp;
+                }
+            }
+            Assert.assertNotNull(filteredIdp, "Added new idps not found in the idp list");
+        } else {
+            Assert.fail("Unable to find added identity provider");
+        }
+    }
+
+    @Test(priority = 5, groups = "wso2.is", description = "test getAllPaginatedIdPsInfo operation with invalid page number")
+    public void testGetPaginatedIdPsWithInvalidPageNumber() {
+
+        try {
+            int pageNumber = -1;
+            List<IdentityProvider> idpList = idpMgtServiceClient.getAllPaginatedIdPsInfo(pageNumber);
+            Assert.assertNull(idpList,
+                    "List of Idps found  while calling getAllPaginatedIdPsInfo() with invalid page number");
+        } catch (Exception e) {
+            Assert.assertEquals(e.getMessage(), invalidPageNumberErrorMessage,
+                    "Error while testing testGetPaginatedIdPsWithInvalidPageNumber() with invalid page number");
```

Comment:
Assert message should convey why the assertion failed. Let;s change as 'Error message of invoking getAllPaginatedIdPsInfo with invalid page number did not match with the expected error message: invalidPageNumberErrorMessage"